core/iwasm/include/mem_alloc_export.h

```diff
@@ -0,0 +1,40 @@
+#ifndef _MEM_ALLOC_EXPORT_H_
+#define _MEM_ALLOC_EXPORT_H_
+
+#include <stdint.h>
+
+/* Memory allocator type */
+typedef enum {
+    /* pool mode, allocate memory from user defined heap buffer */
+    Alloc_With_Pool = 0,
+    /* user allocator mode, allocate memory from user defined
+       malloc function */
+    Alloc_With_Allocator,
+    /* user allocator mode, allocate memory from user defined
+       malloc function with user data */
+    Alloc_With_Allocator_With_Data,
```

Comment:
Should that be moved to the bottom as a last option? Otherwise this change breaks ABI compatibility.

Reply:
Good point, updated